File: OC/爱奇艺播放按钮动画/ViewController.m
//
//  ViewController.m
//  爱奇艺播放按钮动画
//
//  Created by zhangxing on 2017/9/4.
//  Copyright © 2017年 zhangxing. All rights reserved.
//

#import "ViewController.h"
#import "IQiYiPlayButton.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    IQiYiPlayButton *btn = [[IQiYiPlayButton alloc] initWithFrame:CGRectMake(100, 100, 50, 50) state:IQiYiPlayButtonStatePause];
    [btn addTarget:self action:@selector(btnHandle:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
}

- (void)btnHandle:(IQiYiPlayButton *)btn {
    if (btn.buttonState == IQiYiPlayButtonStatePlay) {
        btn.buttonState = IQiYiPlayButtonStatePause;
    }   else {
        btn.buttonState = IQiYiPlayButtonStatePlay;
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Swift/IQiYiPlayButton_Swift/ViewController.swift
//
//  ViewController.swift
//  IQiYiPlayButton_Swift
//
//  Created by zhangxing on 2017/12/8.
//  Copyright © 2017年 zhangxing. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let btn = IQiYiPlayButton.init(frame: CGRect(x: 100.0, y: 100.0, width: 50.0, height: 50.0), state: .pause)
        btn.addTarget(self, action: #selector(btnHandle(btn:)), for: .touchUpInside)
        self.view.addSubview(btn)
    }
    
    @objc func btnHandle(btn: IQiYiPlayButton) {
        //这个判断是必须的
        if btn.isAnimating {
            return
        }
        if btn.buttonState == .play {
            btn.buttonState = .pause
        } else {
            btn.buttonState = .play
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

